test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -169,36 +168,38 @@ public void testInitOfUrlMetadataListerDocIdIsUrlFalse() throws Exception {
   public void testInitOfUrlMetadataListerDocIdIsUrlTrue() throws Exception {
     Map<String, String> moreEntries = new HashMap<String, String>();
     moreEntries.put("db.modeOfOperation", "urlAndMetadataLister");
-    moreEntries.put("db.modeOfOperation.urlAndMetadataLister.columnName", "ur");
     moreEntries.put("docId.isUrl", "true");
     final Config config = createStandardConfig(moreEntries);
     DatabaseAdaptor adaptor = new DatabaseAdaptor();
     adaptor.initConfig(config);
     adaptor.init(TestHelper.createConfigAdaptorContext(config));
   }
 
+  // TODO(bmj): call initConfig() to set defaults for most properties.
   private Config createStandardConfig(Map<String, String> moreEntries) {
     Map<String, String> configEntries = new HashMap<String, String>();
     // driverClass must be specified, but (other than it pointing at a valid
     // class), the value does not matther.
     configEntries.put("db.driverClass",
         "com.google.enterprise.adaptor.database.DatabaseAdaptor");
-    configEntries.put("db.url", "must be set");
-    configEntries.put("db.user", "must be set");
-    configEntries.put("db.password", "must be set");
-    configEntries.put("db.uniqueKey", "must be set:string");
-    configEntries.put("db.everyDocIdSql", "must be set");
-    configEntries.put("db.singleDocContentSqlParameters", "must be set");
-    configEntries.put("db.singleDocContentSql", "must be set");
-    configEntries.put("db.aclSqlParameters", "must be set");
+    configEntries.put("db.url", JdbcFixture.URL);
+    configEntries.put("db.user", "sa");
+    configEntries.put("db.password", "");
+    configEntries.put("db.uniqueKey", "id:int");
```

Comment:
I don't like having "real" defaults here. Having all these fake ones was bad enough, but it's now easy for tests to pass with implicit config, and so the tests are less self-contained. The ones that match initConfig are not a problem (most of the ones from includeAllColumnsAsMetadata down, I think).